docs/source/cpo_trainer.mdx

```diff
@@ -0,0 +1,102 @@
+# CPO Trainer
+
+Contrastive Preference Optimization (CPO) as introduced in the paper [Contrastive Preference Optimization: Pushing the Boundaries of LLM Performance in Machine Translation](https://arxiv.org/pdf/2401.08417.pdf) by Haoran Xu, Amr Sharaf, Yunmo Chen, Weiting Tan, Lingfeng Shen, Benjamin Van Durme, Kenton Murray, and Young Jin Kim. At a high-level CPO  trains models to
+avoid generating adequate but not perfect translations in the Machine Translation (MT) task.
```

Comment:
Does CPO generalise beyond machine translation? My understanding is that Eq. (3) of the paper is a general approximation to the DPO loss, so perhaps we can explain this as follows:

```suggestion
avoid generating adequate, but not perfect translations in Machine Translation (MT) tasks. However, CPO is a general approximation to the DPO loss and can be applied to other domains like chat.
```